File: CsharpDotNet2-client/src/main/CsharpDotNet2/IO/Swagger/Api/TaxConfCFOPApi.cs
using System;
using System.Collections.Generic;
using RestSharp;
using IO.Swagger.Client;
using IO.Swagger.Model;

namespace IO.Swagger.Api
{
    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
    public interface ITaxConfCFOPApi
    {
        /// <summary>
        /// create a new CFOP configuration 
        /// </summary>
        /// <param name="body">The pet JSON you want to post</param>
        /// <returns>InlineResponse201</returns>
        InlineResponse201 CreateCfop (CfopConf body);
        /// <summary>
        /// disable a CFOP. 
        /// </summary>
        /// <param name="code">CFOP Code</param>
        /// <returns></returns>
        void DeleteCfop (string code);
        /// <summary>
        /// get CFOP information. This operation return CFOP configuration 
        /// </summary>
        /// <param name="code">CFOP Code</param>
        /// <returns>CfopConf</returns>
        CfopConf GetCfop (string code);
        /// <summary>
        /// retrive list of CFOP. This operation return CFOP configurations that match with parameters queries 
        /// </summary>
        /// <param name="suffixcode">Identify this CFOP, the CFOP Code has two parts N.XXX where N the prefix is the operation scope type if IN or OUT and if is in state, other state, other country and suffix XXX is the operation type </param>
        /// <param name="date">When informed return valid version configuration for this date </param>
        /// <param name="inactive">return the inactive versions too </param>
        /// <returns>List&lt;CfopConf&gt;</returns>
        List<CfopConf> GetCfopList (string suffixcode, DateTime? date, bool? inactive);
        /// <summary>
        /// update a CFOP configuration 
        /// </summary>
        /// <param name="code">CFOP Code</param>
        /// <param name="taxconfcfop">The pet JSON you want to post</param>
        /// <returns></returns>
        void UpdateCfop (string code, CfopConf taxconfcfop);
    }
  
    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
    public class TaxConfCFOPApi : ITaxConfCFOPApi
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="TaxConfCFOPApi"/> class.
        /// </summary>
        /// <param name="apiClient"> an instance of ApiClient (optional)</param>
        /// <returns></returns>
        public TaxConfCFOPApi(ApiClient apiClient = null)
        {
            if (apiClient == null) // use the default one in Configuration
                this.ApiClient = Configuration.DefaultApiClient; 
            else
                this.ApiClient = apiClient;
        }
    
        /// <summary>
        /// Initializes a new instance of the <see cref="TaxConfCFOPApi"/> class.
        /// </summary>
        /// <returns></returns>
        public TaxConfCFOPApi(String basePath)
        {
            this.ApiClient = new ApiClient(basePath);
        }
    
        /// <summary>
        /// Sets the base path of the API client.
        /// </summary>
        /// <param name="basePath">The base path</param>
        /// <value>The base path</value>
        public void SetBasePath(String basePath)
        {
            this.ApiClient.BasePath = basePath;
        }
    
        /// <summary>
        /// Gets the base path of the API client.
        /// </summary>
        /// <param name="basePath">The base path</param>
        /// <value>The base path</value>
        public String GetBasePath(String basePath)
        {
            return this.ApiClient.BasePath;
        }
    
        /// <summary>
        /// Gets or sets the API client.
        /// </summary>
        /// <value>An instance of the ApiClient</value>
        public ApiClient ApiClient {get; set;}
    
        /// <summary>
        /// create a new CFOP configuration 
        /// </summary>
        /// <param name="body">The pet JSON you want to post</param> 
        /// <returns>InlineResponse201</returns>            
        public InlineResponse201 CreateCfop (CfopConf body)
        {
            
            // verify the required parameter 'body' is set
            if (body == null) throw new ApiException(400, "Missing required parameter 'body' when calling CreateCfop");
            
    
            var path = "/taxconf/cfop";
            path = path.Replace("{format}", "json");
                
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                                postBody = ApiClient.Serialize(body); // http body (model) parameter
    
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.POST, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling CreateCfop: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling CreateCfop: " + response.ErrorMessage, response.ErrorMessage);
    
            return (InlineResponse201) ApiClient.Deserialize(response.Content, typeof(InlineResponse201), response.Headers);
        }
    
        /// <summary>
        /// disable a CFOP. 
        /// </summary>
        /// <param name="code">CFOP Code</param> 
        /// <returns></returns>            
        public void DeleteCfop (string code)
        {
            
            // verify the required parameter 'code' is set
            if (code == null) throw new ApiException(400, "Missing required parameter 'code' when calling DeleteCfop");
            
    
            var path = "/taxconf/cfop/{code}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "code" + "}", ApiClient.ParameterToString(code));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                                    
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.DELETE, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling DeleteCfop: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling DeleteCfop: " + response.ErrorMessage, response.ErrorMessage);
    
            return;
        }
    
        /// <summary>
        /// get CFOP information. This operation return CFOP configuration 
        /// </summary>
        /// <param name="code">CFOP Code</param> 
        /// <returns>CfopConf</returns>            
        public CfopConf GetCfop (string code)
        {
            
            // verify the required parameter 'code' is set
            if (code == null) throw new ApiException(400, "Missing required parameter 'code' when calling GetCfop");
            
    
            var path = "/taxconf/cfop/{code}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "code" + "}", ApiClient.ParameterToString(code));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                                    
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling GetCfop: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling GetCfop: " + response.ErrorMessage, response.ErrorMessage);
    
            return (CfopConf) ApiClient.Deserialize(response.Content, typeof(CfopConf), response.Headers);
        }
    
        /// <summary>
        /// retrive list of CFOP. This operation return CFOP configurations that match with parameters queries 
        /// </summary>
        /// <param name="suffixcode">Identify this CFOP, the CFOP Code has two parts N.XXX where N the prefix is the operation scope type if IN or OUT and if is in state, other state, other country and suffix XXX is the operation type </param> 
        /// <param name="date">When informed return valid version configuration for this date </param> 
        /// <param name="inactive">return the inactive versions too </param> 
        /// <returns>List&lt;CfopConf&gt;</returns>            
        public List<CfopConf> GetCfopList (string suffixcode, DateTime? date, bool? inactive)
        {
            
    
            var path = "/taxconf/cfop";
            path = path.Replace("{format}", "json");
                
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
             if (suffixcode != null) queryParams.Add("suffixcode", ApiClient.ParameterToString(suffixcode)); // query parameter
 if (date != null) queryParams.Add("date", ApiClient.ParameterToString(date)); // query parameter
 if (inactive != null) queryParams.Add("inactive", ApiClient.ParameterToString(inactive)); // query parameter
                                        
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling GetCfopList: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling GetCfopList: " + response.ErrorMessage, response.ErrorMessage);
    
            return (List<CfopConf>) ApiClient.Deserialize(response.Content, typeof(List<CfopConf>), response.Headers);
        }
    
        /// <summary>
        /// update a CFOP configuration 
        /// </summary>
        /// <param name="code">CFOP Code</param> 
        /// <param name="taxconfcfop">The pet JSON you want to post</param> 
        /// <returns></returns>            
        public void UpdateCfop (string code, CfopConf taxconfcfop)
        {
            
            // verify the required parameter 'code' is set
            if (code == null) throw new ApiException(400, "Missing required parameter 'code' when calling UpdateCfop");
            
            // verify the required parameter 'taxconfcfop' is set
            if (taxconfcfop == null) throw new ApiException(400, "Missing required parameter 'taxconfcfop' when calling UpdateCfop");
            
    
            var path = "/taxconf/cfop/{code}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "code" + "}", ApiClient.ParameterToString(code));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                                postBody = ApiClient.Serialize(taxconfcfop); // http body (model) parameter
    
            // authentication setting, if any
            String[] authSettings = new String[] {  };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.PUT, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling UpdateCfop: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling UpdateCfop: " + response.ErrorMessage, response.ErrorMessage);
    
            return;
        }
    
    }
}

File: async-scala-client/src/main/scala/io/swagger/client/api/TaxConfCFOPApi.scala
package io.swagger.client.api

import io.swagger.client.model.CfopConf
import java.util.Date
import io.swagger.client.model.Inline_response_201
import io.swagger.client.model.Message
import io.swagger.client.model.ValidationError
import com.wordnik.swagger.client._
import scala.concurrent.Future
import collection.mutable

class TaxConfCFOPApi(client: TransportClient, config: SwaggerConfig) extends ApiClient(client, config) {

  def createCfop(body: CfopConf)(implicit reader: ClientResponseReader[Inline_response_201], writer: RequestWriter[CfopConf]): Future[Inline_response_201] = {
    // create path and map variables
    val path = (addFmt("/taxconf/cfop"))

    // query params
    val queryParams = new mutable.HashMap[String, String]
    val headerParams = new mutable.HashMap[String, String]

    if (body == null) throw new Exception("Missing required parameter 'body' when calling TaxConfCFOPApi->createCfop")

    val resFuture = client.submit("POST", path, queryParams.toMap, headerParams.toMap, writer.write(body))
    resFuture flatMap { resp =>
      process(reader.read(resp))
    }
  }

  def deleteCfop(code: String)(implicit reader: ClientResponseReader[Unit]): Future[Unit] = {
    // create path and map variables
    val path = (addFmt("/taxconf/cfop/{code}")
        replaceAll ("\\{" + "code" + "\\}",code.toString))

    // query params
    val queryParams = new mutable.HashMap[String, String]
    val headerParams = new mutable.HashMap[String, String]

    if (code == null) throw new Exception("Missing required parameter 'code' when calling TaxConfCFOPApi->deleteCfop")


    val resFuture = client.submit("DELETE", path, queryParams.toMap, headerParams.toMap, "")
    resFuture flatMap { resp =>
      process(reader.read(resp))
    }
  }

  def getCfop(code: String)(implicit reader: ClientResponseReader[CfopConf]): Future[CfopConf] = {
    // create path and map variables
    val path = (addFmt("/taxconf/cfop/{code}")
        replaceAll ("\\{" + "code" + "\\}",code.toString))

    // query params
    val queryParams = new mutable.HashMap[String, String]
    val headerParams = new mutable.HashMap[String, String]

    if (code == null) throw new Exception("Missing required parameter 'code' when calling TaxConfCFOPApi->getCfop")


    val resFuture = client.submit("GET", path, queryParams.toMap, headerParams.toMap, "")
    resFuture flatMap { resp =>
      process(reader.read(resp))
    }
  }

  def getCfopList(suffixcode: Option[String] = None,
      date: Option[Date] = None,
      inactive: Option[Boolean] = None
      )(implicit reader: ClientResponseReader[List[CfopConf]]): Future[List[CfopConf]] = {
    // create path and map variables
    val path = (addFmt("/taxconf/cfop"))

    // query params
    val queryParams = new mutable.HashMap[String, String]
    val headerParams = new mutable.HashMap[String, String]

    suffixcode match {
    case Some(param) => queryParams += "suffixcode" -> param.toString
    case _ => queryParams
    }
    date match {
    case Some(param) => queryParams += "date" -> param.toString
    case _ => queryParams
    }
    inactive match {
    case Some(param) => queryParams += "inactive" -> param.toString
    case _ => queryParams
    }

    val resFuture = client.submit("GET", path, queryParams.toMap, headerParams.toMap, "")
    resFuture flatMap { resp =>
      process(reader.read(resp))
    }
  }

  def updateCfop(code: String,
      taxconfcfop: CfopConf)(implicit reader: ClientResponseReader[Unit], writer: RequestWriter[CfopConf]): Future[Unit] = {
    // create path and map variables
    val path = (addFmt("/taxconf/cfop/{code}")
        replaceAll ("\\{" + "code" + "\\}",code.toString))

    // query params
    val queryParams = new mutable.HashMap[String, String]
    val headerParams = new mutable.HashMap[String, String]

    if (code == null) throw new Exception("Missing required parameter 'code' when calling TaxConfCFOPApi->updateCfop")

    if (taxconfcfop == null) throw new Exception("Missing required parameter 'taxconfcfop' when calling TaxConfCFOPApi->updateCfop")

    val resFuture = client.submit("PUT", path, queryParams.toMap, headerParams.toMap, writer.write(taxconfcfop))
    resFuture flatMap { resp =>
      process(reader.read(resp))
    }
  }


}
